devito/mpi/halo_scheme.py

```diff
@@ -287,7 +287,11 @@ def omapper(self):
         ranks*, so the output of this method is guaranteed to be consistent
         across *all MPI ranks*.
         """
-        items = [((d, CENTER), (d, LEFT), (d, RIGHT)) for d in self.dimensions]
+
+        dimensions = OrderedSet(*sorted(list(self.dimensions),
```

Comment:
why not just `filter_sorted(self.dimension)` ?

Reply:
Sure